src/CSI.IBTA.BenefitsService/Services/UserBalanceService.cs

```diff
@@ -0,0 +1,61 @@
+﻿using CSI.IBTA.BenefitsService.Interfaces;
+using CSI.IBTA.DataLayer.Interfaces;
+using Microsoft.EntityFrameworkCore;
+using CSI.IBTA.Shared.Entities;
+using CSI.IBTA.Shared.DTOs;
+using CSI.IBTA.Shared.DTOs.Errors;
+using System.Net;
+
+namespace CSI.IBTA.BenefitsService.Services
+{
+    internal class UserBalanceService : IUserBalanceService
+    {
+        private readonly IBenefitsUnitOfWork _benefitsUnitOfWork;
+
+        public UserBalanceService(IBenefitsUnitOfWork benefitsUnitOfWork)
+        {
+            _benefitsUnitOfWork = benefitsUnitOfWork;
+        }
+
+        public async Task<GenericResponse<decimal>> GetCurrentBalanceForPlan(int planId)
+        {
+            var plan = await _benefitsUnitOfWork.Plans
+                .Include(c => c.Package)
+                .FirstOrDefaultAsync(x => x.Id == planId);
+
+            if(plan == null) return new GenericResponse<decimal>(HttpErrors.ResourceNotFound, 0);
+            if(!plan.Package.IsActive) return new GenericResponse<decimal>(new HttpError("This plan is not active yet", HttpStatusCode.BadRequest), 0);
+
+            var package = plan.Package;
+            var negativeTransactions = await _benefitsUnitOfWork.Claims
+                .Find(x => x.PlanId == plan.Id && x.Status == ClaimStatus.Approved);
```

Comment:
I missed it previous review. 
Claim != Transaction

Claim is a payment **request**, when transaction is an actual funds transfer. 
The code should calculate sum of actual payments ( bank transactions) between employer -> consumer ( +) and consumer -> admin ( - ).

It should be discussed, may be what you have in the code is good enough for stakeholders.

Reply:
@VidasBui  - I have no chances to join for today's call 3/15, so please speak to Sergey about balance calculation and if he is OK to see sum of claims instead of sum of transactions. 

Approving your PR for now